src/GafferOSL/OSLExpressionEngine.cpp

```diff
@@ -219,12 +219,30 @@ class RendererServices : public OSL::RendererServices
 // OSLExpressionEngine
 //////////////////////////////////////////////////////////////////////////
 
+namespace
+{
+
 typedef pair<string, string> Replacement;
 bool replacementGreater( const Replacement &lhs, const Replacement &rhs )
 {
 	return lhs.first.size() > rhs.first.size();
 }
 
+void replaceAll( std::string &s, vector<Replacement> &replacements )
```

Comment:
If this were a general purpose function in the public API, I would be very wary of the side-effect on `replacements`, but I think it is probably OK here for this limited use? I didn't want to do an unnecessary copy, and I did want to pull the replacement logic into one function so I could explain it in one place.

Reply:
I don't see anything wrong with it for this limited case, but it probably would be good to add an explicit comment here, and possibly at the call site as well.

Reply:
Added a comment in 524d301.